### ISSUE TITLE
ORecordId.equals() should accept and evaluate String

### ISSUE THREAD

**Issue by myukselen** [orientechnologies/orientdb#8254]
## Expected behavior  
From gremlin console, it is possible to get vertex by string and it should be possible to check its id with a string again.

## Actual behavior  
Although string Record Id is used to find the vertex, its equality check does not try to convert string to ORecordId with its string constructor. It simply returns false.

## Steps to reproduce  
in gremlin console with remote connection to demodb:
```
gremlin> :> g.V('33:7').id().is('#33:7')
gremlin> :> g.V('33:7').id().is('33:7')
```

it should return the vertex ```#33:7```

**Comment by maggiolo00** [comment]
Hi @myukselen 

we cannot change the default implementation of `ORecordId.equals()` for compatibility issues.

Let me check if we can do something in the implementation of Tinkerpop

Thanks

**Comment by myukselen** [comment]
Hello @maggiolo00 

Thanks for your feedback. I wanted to point a problem in usability perspective for Gremlin users.

I respect as the design choices are yours to manage. I implemented my suggestion on the source and evaluated further on how can I utilize OrientDB from Python. I ended up in other problems.

Thanks for the opportunity OrientDB Team provides by allowing us to study/change the code we use.
